lib/fathead/sqlalchemy/parse.py

```diff
@@ -0,0 +1,339 @@
+
+        return txt
+
+
+    def write_to_file(self,filename,elements):
+        """
+            Function to append a given statement to
+            the given file
+        """
+        # Append to the file
+        currFile = open(filename,'a')
+
+        for e in elements:
+            currFile.write(elements[e])
+
+        currFile.close()
+
+    def remove_newline(self,text):
+        """
+            Function to remove '\n'
+            from the start of a sentence
+        """
+        t_list = text.split('\\n')
+        txt = ''
+        
+        for k in t_list:
+            if k!='' and k!=' ':
+                txt += k+' '
+
+        return txt
+
+    def get_table_contents(self,table):
+        """
+            Function to return the 
+            contents of a table after
+            checking for code snippets
+        """
+
+        if table.find('div',{'class':'highlight-default'}) is None:
+            return table.text # If table doesn't have snippets then return entire text
+        
+        tbody = table.find('tbody')
+
+        th =  tbody.find('th')
+        text = '\\n\\n'+th.text
+        td = tbody.find('td')
+        ul = td.find('ul')
+        text += '<ul>{}</ul>' # Fetch the elements and insert them into unordered list
+        lis=''
+
+        for li in ul:
+            cur = '<li>{}</li>\\n'
+            txt=''
+            for e in li:
+                try:
+                    if e.name=='div' and 'highlight-default' in e['class']:
+                        txt+='<pre><code>{}</code></pre>'.format(e.text)
+                    else:
+                        txt+=e.text
+                except:
+                    if str(type(e))=="<class 'bs4.element.NavigableString'>":
+                        txt+=e.string
+
+            cur=cur.format(self.replace_unicodes(txt))
+            lis+=cur
+        text = text.format(lis)
+
+        return text
+                
+
+        
+    def generate_pages(self,page_name):
+        """
+            Function to parse the source code 
+            of the internals page and write the 
+            required functions onto output.txt
+        """
+        
+        elements = self.elements
+        repeated = self.repeated
+        description = self.description
+        title_chck = self.titles
+
+        if page_name=='internals':
+            soup = self.internals_page
+        elif page_name=='events':
+            soup = self.events_page
+        else:
+            soup = self.exceptions_page
+
+        # Find all 'dl' tags in the page
+        dls = soup.find_all('dl')
+
+        for k in dls:
+
+            curr_dd = k.find('dd') # Get the first 'dd' tag in the current 'dl' tag
+            
+            # The required components of each element
+            section = '<section class="prog__container">{}</section>'
+            code = '<pre><code>{}</code></pre>'
+            desc='<p>{}</p>'
+            header = '<span class="prog__sub">{}</span>'
```

Comment:
@shravan97 I think this might look better as a `<pre><code>` so the text doesn't wrap? We could also give it a heading like "Usage", or "Definition" if needed. It would look like this:

>**properties.RelationshipProperty.Comparator.contains**
>
>`contains(other, **kwargs)`
>
>Return a simple expression that tests a collection for containment of a particular item.contains is only valid for a collection, i.e. a relationship that implements one-to-many or many-to-many with uselist=True.When used in a simple one-to-many context, an expression like: